src/AzureClient/Mocks/MockCloudJob.cs

```diff
@@ -7,9 +7,89 @@
 using Azure.Quantum.Jobs.Models;
 using System;
 using System.IO;
+using System.Collections.Generic;
 
 namespace Microsoft.Quantum.IQSharp.AzureClient
 {
+    public class MockCostEstimate
+    {
+        public MockCostEstimate(string currencyCode, IReadOnlyList<UsageEvent> events, float? estimatedTotal)
+        {
+            this.InternalCostEstimate = MockHelper.CreateWithNonPublicConstructor<CostEstimate>(currencyCode, events, estimatedTotal);
+        }
+
+        public string CurrencyCode
+        {
+            get => this.InternalCostEstimate.CurrencyCode;
+            set => this.InternalCostEstimate.SetReadOnlyProperty<CostEstimate>(nameof(CurrencyCode), value);
+        }
+
+        public IReadOnlyList<UsageEvent> Events
+        {
+            get => this.InternalCostEstimate.Events;
+        }
+
+        public float? EstimatedTotal
+        {
+            get => this.InternalCostEstimate.EstimatedTotal;
+            set => this.InternalCostEstimate.SetReadOnlyProperty<CostEstimate>(nameof(EstimatedTotal), value);
+        }
+
+        public CostEstimate InternalCostEstimate { get; }
+    }
+
+    internal class MockJobDetails : JobDetails
+    {
+        public MockJobDetails(string containerUri, string inputDataFormat, string providerId, string target)
+            : base(containerUri: containerUri, 
+                   inputDataFormat: inputDataFormat, 
+                   providerId: providerId, 
+                   target: target)
+        {
+        }
+
+        private MockCostEstimate? costEstimate;
+        public new MockCostEstimate? CostEstimate
+        {
+            get => this.costEstimate;
+            set
+            {
+                this.costEstimate = value;
+                this.SetReadOnlyProperty<JobDetails>(nameof(CostEstimate), value?.InternalCostEstimate);
+            }
```

Comment:
For eventual .NET 6, we may want to consider using the new `field` keyword in C# 10 to simplify this some.

```csharp
        public new MockCostEstimate? CostEstimate
        {
            get;
            set
            {
                field = value;
                this.SetReadOnlyProperty<JobDetails>(nameof(CostEstimate), value?.InternalCostEstimate);
            }
        }
```

Reply:
Since the unit test project is still targeting `netcoreapp3.1`, I'll leave it like this for now.